claimManagement/src/main/java/org/openimis/imisclaims/usecase/FetchInsureeInquire.java

```diff
@@ -0,0 +1,125 @@
+    @Nullable
+    private String getPhotoPath(@NonNull List<GetInsureeInquireQuery.Photo> photos) {
+        for (GetInsureeInquireQuery.Photo photo : photos) {
+            String filename = photo.filename();
+            if (filename != null) {
+                String folder = photo.folder();
+                if (folder != null) {
+                    folder = folder.replace('\\', '/');
+                    if (!folder.endsWith("/")) {
+                        folder += "/";
+                    }
+                    return folder + filename;
+                }
+                return filename;
+            }
+        }
+
+        return null;
+    }
+
+    @Nullable
+    private byte[] getPhotoBytes(@NonNull List<GetInsureeInquireQuery.Photo> photos) {
+        for (GetInsureeInquireQuery.Photo photo : photos) {
+            String photoBase64 = photo.photo();
+            if (photoBase64 != null) {
+                return Base64.decode(photoBase64, Base64.DEFAULT);
+            }
+        }
+
+        return null;
+    }
+
+    @NonNull
+    private Policy toPolicy(@NonNull GetInsureeInquireQuery.Edge1 edge) {
+        GetInsureeInquireQuery.Policy policy = Objects.requireNonNull(edge.node()).policy();
+        GetInsureeInquireQuery.Product product = policy.product();
+        return new Policy(
+                /* code = */ product.code(),
+                /* name = */ product.name(),
+                /* value = */ policy.value(),
+                /* expiryDate = */ policy.expiryDate(),
+                /* status = */ intAsStatus(policy.status()),
+                /* deductibleType = */ product.deductible(),
+                /* deductibleIp = */ product.deductibleIp(),
+                /* deductibleOp = */ product.deductibleOp(),
+                /* ceilingIp = */ product.ceilingIp(),
+                /* ceilingOp = */ product.ceilingOp(),
+                /* antenatalAmountLeft = */ product.maxAmountAntenatal(),
+                /* consultationAmountLeft = */ product.maxAmountConsultation(),
+                /* deliveryAmountLeft = */ product.maxAmountDelivery(),
+                /* hospitalizationAmountLeft = */ product.maxAmountHospitalization(),
+                /* surgeryAmountLeft = */ product.maxAmountSurgery(),
+                /* totalAdmissionsLeft = */ product.maxNoHospitalization(),
+                /* totalAntenatalLeft = */ product.maxNoAntenatal(),
+                /* totalConsultationsLeft = */ product.maxNoConsultation(),
+                /* totalDeliveriesLeft = */ product.maxNoDelivery(),
+                /* totalSurgeriesLeft = */ product.maxNoSurgery(),
+                /* totalVisitsLeft = */ product.maxNoVisits()
+        );
+    }
+
+    /**
+     * <a href="https://github.com/openimis/database_ms_sqlserver/blob/main/sql/stored_procedures/uspAPIGetCoverage.sql#L48C3-L48C101">Values for the status in the stored procedure</a>
+     */
+    @NonNull
+    private Policy.Status intAsStatus(@Nullable Integer integer) {
```

Comment:
I'd put that in the enum class `Policy.Status`.

Reply:
I don't want my domain object to know how it is mapped into/from the API response. Therefore, that's why it's here and not in the domain object.

Reply:
we discussed that live on Friday. So :+1: